packages/flutter_test/lib/src/_goldens_io.dart

```diff
@@ -125,26 +125,28 @@ class LocalComparisonOutput {
   /// Writes out diffs from the [ComparisonResult] of a golden file test.
   ///
   /// Will throw an error if a null result is provided.
-  void generateFailureOutput(
+  Future<void> generateFailureOutput(
     ComparisonResult result,
     Uri golden,
     Uri basedir, {
     String key = '',
-  }) {
+  }) async {
     String additionalFeedback = '';
     if (result.diffs != null) {
       additionalFeedback = '\nFailure feedback can be found at '
         '${path.join(basedir.path, 'failures')}';
       final Map<String, Image> diffs = result.diffs.cast<String, Image>();
-      diffs.forEach((String name, Image image) {
+      for (final MapEntry<String, Image> entry in diffs.entries) {
         final File output = getFailureFile(
-          key.isEmpty ? name : name + '_' + key,
+          key.isEmpty ? entry.key : entry.key + '_' + key,
           golden,
           basedir,
         );
         output.parent.createSync(recursive: true);
-        output.writeAsBytesSync(encodePng(image));
-      });
+        final ByteData pngBytes =
+            await entry.value.toByteData(format: ImageByteFormat.png);
```

Comment:
```suggestion
        final ByteData pngBytes = await entry.value.toByteData(format: ImageByteFormat.png);
```

Reply:
This does push the line over 80 characters - is there a different way I should do the line wrapping?